db_test.go

```diff
@@ -695,6 +695,32 @@ func TestWALFlushedOnDBClose(t *testing.T) {
 	testutil.Equals(t, []string{"labelvalue"}, values)
 }
 
+func TestWALSegmentSizeOption(t *testing.T) {
+	options := *DefaultOptions
+	options.WALSegmentSize = 2 * 32 * 1024 //one segment file must contains 2 pages max (32 Kb each)
+	db, close := openTestDB(t, &options)
+	defer close()
+	app := db.Appender()
+	for i := int64(0); i < 155; i++ {
+		_, err := app.Add(labels.Labels{{"wal", "size"}}, i, rand.Float64())
+		testutil.Ok(t, err)
+		testutil.Ok(t, app.Commit())
+	}
+
+	walFiles := map[string]int64{}
+	files, err := ioutil.ReadDir(db.Dir() + "/wal")
+	if err != nil {
+		t.Fatalf("Error while reading wal directory: %v", err)
+	}
+	for _, f := range files {
+		walFiles[f.Name()] = f.Size()
+	}
+
+	testutil.Equals(t, int64(options.WALSegmentSize), walFiles["00000000"], "Wal file 00000000 must size WALSegmentSize")
+	testutil.Equals(t, int64(options.WALSegmentSize), walFiles["00000001"], "Wal file 00000001 must size WALSegmentSize")
+	testutil.Assert(t, walFiles["00000002"] > 0, "Wal file 00000002 must have been created")
```

Comment:
I think I managed to simplify this part a bit and removed the need for hard coded filenames strings.

```
testutil.Assert(t, len(files) > 1, "current WALSegmentSize should result in more then a WAL single file.")

for i, f := range files {
	if len(files)-1 != i {
		testutil.Equals(t, int64(options.WALSegmentSize), f.Size(), "WAL file size doesn't match WALSegmentSize option, filename: %v", f.Name())
		continue
	}
	testutil.Assert(t, int64(options.WALSegmentSize) > f.Size(), "last WAL file size is not smaller than the WALSegmentSize option, filename: %v", f.Name())
}
```

Reply:
thanks, it's done